File: playbook-java/src/main/java/io/infinitic/playbook/java/asyncMethod/Start.java
package io.infinitic.playbook.java.asyncMethod;

import io.infinitic.clients.InfiniticClient;
import io.infinitic.playbook.java.asyncMethod.workflows.AsyncMethodWorkflow;

import java.util.Set;

public class Start {
    static final String asyncMethodTag = "asyncMethod";

    public static void main(String[] args) {
        try(InfiniticClient client = InfiniticClient.fromConfigResource("/infinitic.yml")) {
            // create a stub for AsyncMethodWorkflow, with tag "promises"
            AsyncMethodWorkflow instance = client.newWorkflow(AsyncMethodWorkflow.class, Set.of(asyncMethodTag));

            // start new workflow
            client.dispatchVoid(instance::run);
        }
    }
}

File: playbook-kotlin/src/main/kotlin/io/infinitic/playbook/kotlin/asyncMethod/Start.kt
package io.infinitic.playbook.kotlin.asyncMethod

import io.infinitic.clients.InfiniticClient
import io.infinitic.playbook.kotlin.asyncMethod.workflows.AsyncMethodWorkflow

internal const val asyncMethodTag = "asyncMethod"

fun main() {
    InfiniticClient.fromConfigResource("/infinitic.yml").use { client ->
        // create a stub for PromisesWorkflow, with tag "promises"
        val workflow = client.newWorkflow(AsyncMethodWorkflow::class.java, setOf(asyncMethodTag))

        // start new workflow
        client.dispatch(workflow::run)
    }
}
